### PR TITLE
move geo and geo_lookup from PX4 Firmware to ECL

### PR DESCRIPTION
**Pull request by dagar**

### CODE REVIEW

File: geo_lookup/geo_mag_declination.cpp [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+/** set this always to the sampling in degrees for the table below */
+#define SAMPLING_RES		10.0f
+#define SAMPLING_MIN_LAT	-60.0f
+#define SAMPLING_MAX_LAT	60.0f
+#define SAMPLING_MIN_LON	-180.0f
+#define SAMPLING_MAX_LON	180.0f
+
+#define constrain(val, min, max) (val < min) ? min : ((val > max) ? max : val)
+
+static const int8_t declination_table[13][37] = \
```
Comment (priseborough) [inline geo_lookup/geo_mag_declination.cpp]: This table and the citation for the data source needs to be be updated annually. See https://github.com/priseborough/Firmware/tree/pr-geoMag

Comment (priseborough) [reply geo_lookup/geo_mag_declination.cpp]: I can update the table data for this PR if you wish.

Comment (dagar) [reply geo_lookup/geo_mag_declination.cpp]: Yes please. That means PX4 (including EKF2 in PX4) was using the old table.

**Comment by dagar** [comment]
@priseborough I pulled your Firmware side magnetism table changes (https://github.com/priseborough/Firmware/commit/75e712f2769ef109a458d2e6c58ce2390045400e) into this branch.

**Comment by dagar** [comment]
Can be tested with this Firmware branch. https://github.com/PX4/Firmware/pull/9105

**Comment by priseborough** [comment]
Thanks for that. I will look through and test.

**Pull request merged**